### PR TITLE
add native support for influxDB 2.0 / Dashboard for influxDB 2.0

### PR DESCRIPTION
**Pull request by derpeter**
Hey, 

here a first try to extend the starlink-grpc-tools to work natively with inlfuxDB 2.0

### CODE REVIEW

File: dish_grpc_influx2.py [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+                       default=BUCKET_DEFAULT,
+                       help="Bucket name to use, default: " + BUCKET_DEFAULT)
+    group.add_argument("-O", "--org", help="Organisation name")
+    group.add_argument("-k",
+                       "--skip-query",
+                       action="store_true",
+                       help="Skip querying for prior sample write point in bulk mode")
+    group.add_argument("-C",
+                       "--ca-cert",
+                       dest="verify_ssl",
```
Comment (sparky8512) [inline dish_grpc_influx2.py:74]: This option did not work for me. It looks like the API changed so that CA cert file is no longer overloaded with "verify_ssl" the way it was in the older InfluxDB client and that there is now a separate "ssl_ca_cert" kwarg.

File: dish_grpc_influx2.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+                       help="Enable SSL/TLS using specified CA cert to verify broker",
```
Comment (sparky8512) [inline dish_grpc_influx2.py:75]: Should probably be "server" instead of "broker". Looks like a copy/paste error on my part in dish_grpc_influx.py (which I will fix).

File: dish_grpc_influx2.py [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+    opts = dish_common.run_arg_parser(parser, need_id=True)
+
+    opts.icargs = {}
+    for key in ["url", "token", "bucket", "org", "verify_ssl"]:
+        val = getattr(opts, key)
+        if val is not None:
+            opts.icargs[key] = val
+
+    if opts.verify_ssl is not None:
+        opts.icargs["ssl"] = True
```
Comment (sparky8512) [inline dish_grpc_influx2.py:117]: Another API change here, I think. kwarg "ssl" appears to be ignored. Probably needs to be encoded into URL instead.

File: dish_grpc_influx2.py [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+            |> filter(fn: (r) => r["_measurement"] == "{3}")
+            |> filter(fn: (r) => r["_field"] == "counter")
+            |> last()
+            |> yield(name: "last")
+            '''.format(opts.bucket, str(start), str(end), BULK_MEASUREMENT)
+        )
+        if result:
+            counter = result[0].records[0]['_value']
+            timestamp = result[0].records[0]['_time']
+            timestamp = time.mktime(timestamp.timetuple())
```
Comment (sparky8512) [inline dish_grpc_influx2.py:173]: In my testing, this appeared to have converted wrong. This should be seconds since epoch as a UTC time (actually, it should be whatever time.time() returns, but I have yet to encounter a system where that wasn't UTC), and this seems to be offsetting by local time offset, instead.

While this isn't critical functionality, as it's just making sure there isn't a +/-1 second offset from prior logged bulk history values, if it's going to be included, it should be corrected. timestamp.timestamp() seems to give the correct value.

File: dish_grpc_influx2.py [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+            if counter and timestamp:
+                return int(counter), int(timestamp)
+    except TypeError as e:
```
Comment (sparky8512) [inline dish_grpc_influx2.py:176]: This probably isn't needed any more with the InfluxDB 2.x client, but I expect it is harmless.

File: dish_grpc_influx2.py [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+
+    if "verify_ssl" in opts.icargs and not opts.icargs["verify_ssl"]:
+        # user has explicitly said be insecure, so don't warn about it
+        warnings.filterwarnings("ignore", message="Unverified HTTPS request")
+
+    signal.signal(signal.SIGTERM, handle_sigterm)
+    try:
+        gstate.influx_client = InfluxDBClient(**opts.icargs)
+    except TypeError as _err:
+        print('Error while creating influx client: ' + str(_err))
```
Comment (sparky8512) [inline dish_grpc_influx2.py:305]: This seems like it would continue while leaving gstate.influx_client unset.

It really should either sys.exit or re-throw the exception. Or just remove the try clause and let the exception abort the script run. The InfluxDB 1.x script only caught TypeError here in an effort to hack around old versions of the influxdb module that didn't support another hack-around which wouldn't be relevant when using the InfluxDB 2.x client module.

**Review by sparky8512 — APPROVED** [review]
This looks good. Thanks for submitting it.

I have a few comments, but I'm going to go ahead and pull it in once I'm done testing, since it's a separate script and all my comments are about mostly corner cases.

**Comment by derpeter** [comment]
Thanks for the detailed feedback. I completly forgot to test with SSL enabled as i run it on the same host. I will try to resolve the comments as soon as i can.
